src/actions/authentication.js

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios';
+import jwt_decode from 'jwt-decode';
+import { GET_ERRORS, SET_CURRENT_USER } from './types';
+import setAuthToken from '../setAuthToken';
+
+
+export const registerUser = (user, history) => dispatch => {
+    axios.post('/api/users/register', user)
+            .then(res => history.push('/login'))
+            .catch(err => {
+                dispatch({
+                    type: GET_ERRORS,
+                    payload: err.response.data
+                });
+            });
+}
+
+export const loginUser = (user) => dispatch => {
```

Comment:
you can add LOGIN_PENDING, LOGIN_SUCESS and LOGIN_ERROR actions in order to control redux from redux dev tools and add some loading icon